Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -95,5 +128,310 @@ public static class CustomEntitySearch
+                                        {
+                                            currMismatch += potWordMismatch;
+                                            for (int i = 0; i < offsetWord; i++)
+                                                wordFound.Write(text[currTextCharIndex + i]);
+                                            currTextCharIndex += offsetWord;
+                                            currWordCharIndex += offsetWord;
+                                        }
+                                        else if ((potWordMismatch > potTextMismatch && potTextMismatch != 0) || potWordMismatch == 0)
+                                        {
+                                            currMismatch += 1;
+                                            wordFound.Write(text[currTextCharIndex]);
+                                            currTextCharIndex++;
+                                        }
+                                        else
+                                        {
+                                            currMismatch += potWordMismatch;
+                                            currWordCharIndex += offsetWord;
+                                        }
+                                    }
+                                    else if (trueTextComp)
+                                    {
+                                        currMismatch += 1;
+                                        wordFound.Write(text[currTextCharIndex]);
+                                        currTextCharIndex++;
+                                    }
+                                    else if (trueWordComp)
+                                    {
+                                        currMismatch += potWordMismatch;
+                                        currWordCharIndex += offsetWord;
+                                    }
+                                    else
+                                    {
+                                        currMismatch += 1;
+                                        currWordCharIndex++;
+                                        if (!(currWordCharIndex >= wordCharArray.Length && textCharArray[currTextCharIndex].IsDelineating()) && currMismatch <= leniency)
+                                        {
+                                            wordFound.Write(text[currTextCharIndex]);
+                                            currTextCharIndex++;
+                                        }
+                                    }
+
+                                }
+                            }
+                        }
+                        else
+                        {
+                            int initialOffsetIndex = currTextCharIndex - wordFound.ToString().Length;
+                            currWordCharIndex = 0;
+
+                            if (bestMismatchPreWhitespace > -1 && whitespaceOffset != initialOffsetIndex)
```

Comment:
>(bestMismatchPreWhitespace > -1 && [](start = 31, length = 34)

why check that there is a bestMismatchPreWhitespace in this case? Regardless of if a reasonable match was found, you should still proceed to the next whitespace?

same for line ~380

Reply:
This check was made for the "hello hello hello" text and "hello hello" word case